test/handlers/images_test.py

```diff
@@ -750,8 +759,110 @@ def test_options_write(srv, client, tmpdir):
     assert set(res.getheader("allow").split(',')) == allows
     options = json.loads(res.read())
     assert set(options["features"]) == ALL_FEATURES
+
+    # File backend specific options.
+    assert options["max_readers"] == srv.config.daemon.max_connections
+    assert options["max_writers"] == 1
+    assert "transfer_format" not in options
+    assert "virtual_size" not in options
+
+
+@pytest.mark.parametrize("fmt", ["raw", "qcow2"])
+def test_options_nbd_read(srv, client, tmpdir, nbd_server, fmt):
+    # Create disk.
+    size = GiB
+    disk = str(tmpdir.join(f"disk.{fmt}"))
+    qemu_img.create(disk, fmt, size=size)
+
+    # Start nbd server exporting the disk.
+    nbd_server.image = disk
+    nbd_server.fmt = fmt
+    nbd_server.read_only = True
+    nbd_server.start()
+
+    # Add ticket using nbd server url.
+    ticket = testutil.create_ticket(
+        url=nbd_server.sock.url(), size=size, ops=["read"])
+    srv.auth.add(ticket)
+
+    # Get OPTIONS.
+    res = client.options("/images/" + ticket["uuid"])
+    allows = {"OPTIONS", "GET"}
+    assert res.status == 200
+    assert set(res.getheader("allow").split(',')) == allows
+    options = json.loads(res.read())
+    assert set(options["features"]) == BASE_FEATURES
+
+    # NBD backend specific options.
+    assert options["max_readers"] == srv.config.daemon.max_connections
+    assert options["max_writers"] == srv.config.daemon.max_connections
+    assert options["transfer_format"] == "raw"
+    assert options["virtual_size"] == size
+
+
+@pytest.mark.parametrize("fmt", ["raw", "qcow2"])
+def test_options_nbd_write(srv, client, tmpdir, nbd_server, fmt):
+    # Create disk.
+    size = GiB
+    disk = str(tmpdir.join(f"disk.{fmt}"))
+    qemu_img.create(disk, fmt, size=size)
+
+    # Start nbd server exporting the disk.
+    nbd_server.image = disk
+    nbd_server.fmt = fmt
+    nbd_server.start()
+
+    # Add ticket using nbd server url.
+    ticket = testutil.create_ticket(
+        url=nbd_server.sock.url(), size=size, ops=["write"])
+    srv.auth.add(ticket)
+
+    # Get OPTIONS.
+    res = client.options("/images/" + ticket["uuid"])
+    allows = {"OPTIONS", "GET", "PUT", "PATCH"}
+    assert res.status == 200
+    assert set(res.getheader("allow").split(',')) == allows
+    options = json.loads(res.read())
+    assert set(options["features"]) == ALL_FEATURES
+
+    # NBD backend specific options.
     assert options["max_readers"] == srv.config.daemon.max_connections
-    assert options["max_writers"] == 1  # Using file backend.
+    assert options["max_writers"] == srv.config.daemon.max_connections
+    assert options["transfer_format"] == "raw"
+    assert options["virtual_size"] == size
+
+
+@pytest.mark.parametrize("size", [
+    KiB,
+    512 * KiB,
+    MiB,
+    512 * MiB,
+    GiB,
+    512 * GiB,
+    TiB,
+    8 * TiB
+])
+def test_options_nbd_size_check(srv, client, tmpdir, nbd_server, size):
```

Comment:
I think this is already covered by the nbd_{read,write} tests, and testing different sizes do not add enough value to run this test multiple times.